src/de/seq.rs

```diff
@@ -1,18 +1,74 @@
 use crate::de::{DeError, DeEvent, Deserializer, XmlRead};
 use crate::events::BytesStart;
+use crate::reader::Decoder;
 use serde::de::{self, DeserializeSeed};
+#[cfg(not(feature = "encoding"))]
+use std::borrow::Cow;
 
+/// Check if tag `start` is included in the `fields` list. `decoder` is used to
+/// get a string representation of a tag.
+///
+/// Returns `true`, if `start` is not in the `fields` list and `false` otherwise.
+pub fn is_unknown(
```

Comment:
Could this be named more descriptively e.g. `is_unknown_{field|tag|element}`?

Reply:
"unknown" also sounds a little strange in this context since we theoretically do known the names of the enum variants?

re:

> 
>     enum Choice { one, two, three }
>     struct Root {
>         #[serde(rename = "$value")]
>         item: [Choice; 3],
>         node: (),
>     }
> 
> and this struct should be able to deserialized from
> 
>     <root>
>         <one/>
>         <two/>
>         <three/>
>         <node/>
>     </root>

Maybe `tag_maps_to_field`?  (That name sounds a bit wrong as well, so if you prefer the former names I'm fine with that)

Reply:
I chose `is_unknown` simple in contrary to `is_known` (i.e. if tag is listed in `fields` it is known for a map), but because each usage of `is_known()` would be `!is_known()` I inverted it. The method does not have any specific cognitive load, so just `not_in` should be enough.